timemachine/fe/chiral_utils.py

```diff
@@ -484,3 +493,336 @@ def make_chiral_flip_heatmaps(simulation_result, atom_map):
     assert mol_a_chiral_conflicts.shape == (len(simulation_result.frames), len(simulation_result.frames[0]))
 
     return mol_a_chiral_conflicts, mol_b_chiral_conflicts
+
+
+def get_num_connected_components(num_atoms: int, bonds: Collection[Tuple[int, int]]) -> int:
+    g = nx.Graph()
+    g.add_nodes_from(range(num_atoms))
+    g.add_edges_from(bonds)
+    return len(list(nx.connected_components(g)))
+
+
+def canonicalize_chiral_atom_idxs(idxs):
+    i, j, k, l = idxs
+    rotations = [(j, k, l), (l, j, k), (k, l, j)]
+    jj, kk, ll = min(rotations)
+    return i, jj, kk, ll
+
+
+# (ytz): TODO: Refactor this so the HarmonicBond setup is done in single_topology.py in the future and
+# isolate functionality in this module to chirality-specific tasks.
+def setup_end_state_harmonic_bond_and_chiral_potentials(
+    ff, mol_a, mol_b, core, a_to_c, b_to_c
+) -> Tuple[BoundPotential[HarmonicBond], BoundPotential[ChiralAtomRestraint], BoundPotential[ChiralBondRestraint]]:
+    """
+    Setup end-state potentials to verify chiral correctness for mol_a with dummy atoms of mol_b attached. The mapped indices will correspond
+    to the alchemical molecule with dummy atoms. Note that the bond, chiral atom and chiral bond idxs are canonicalized.
+
+    This code is identical to setup_end_state, but only handles chiral potentials to be used to quickly verify
+    core mappings in verify_chiral_consistency_of_core
+
+    Parameters
+    ----------
+    ff: forcefield.Forcefield
+        Forcefield used to parameterize the molecule
+
+    mol_a: Chem.Mol
+        Fully interacting molecule
+
+    mol_b: Chem.Mol
+        Molecule providing the dummy atoms.
+
+    core: list of 2-tuples
+        Each pair is an atom mapping from mol_a into mol_b
+
+    a_to_c: dict or array, supports []
+        mapping from a into a common core idx
+
+    b_to_c: dict or array, supports []
+        mapping from b into a common core idx
+
+    Returns
+    -------
+    Tuple of bound HarmonicBond, ChiralAtomRestraint and ChiralBondRestraint
+        The potentials involved in checking the correctness and validity of chirality at endstates.
+
+    """
+    all_dummy_bond_idxs, all_dummy_bond_params = [], []
+    all_dummy_chiral_atom_idxs, all_dummy_chiral_atom_params = [], []
+
+    from timemachine.fe.dummy import convert_bond_list_to_nx, generate_dummy_group_assignments, get_romol_bonds
```

Comment:
nit: move import to top (unless intentional?)

Reply:
willfix